### DISCUSSION TITLE
Support for aggregates/grouping

### DISCUSSION THREAD

**Discussion by aaronleopold** [Brendonovich/prisma-client-rust#74] [Ideas]
Hi hello! I know your focus isn't on the prisma client right now, as you mentioned in the lastest release, but I was wondering if there are any future plans to support aggregate functions and group by clauses? `_.query_raw` does satisfy my need currently, was just curious if this was considered.

**Comment by Brendonovich** [comment]
Definitely! I thought I had an issue for it but I guess not haha. It's a killer feature of Prisma but the tricky thing is how to actually do it.
In Typescript the return type of aggregate queries can be adjusted depending on the arguments passed in, making them super nice to use. In Rust we don't really have that luxury. Prisma Client Go discussed parsing the Go AST in https://github.com/prisma/prisma-client-go/issues/9, and maybe we could do something similar with macros, but I've been thinking about ways to do it for a while and still have no idea.
If you have any ideas I'm open to trying them out, it'd be super useful!

**Reply by aaronleopold** [reply]
If I think of anything I can let you know, thanks!!
